### Importación de datos



In [1]:

pip install pandas pyarrow

In [2]:
pip install plotly

In [3]:
pip install nbformat

In [4]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url, engine="pyarrow")
tienda2 = pd.read_csv(url2, engine="pyarrow")
tienda3 = pd.read_csv(url3, engine="pyarrow")
tienda4 = pd.read_csv(url4, engine="pyarrow")

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#1. Análisis de facturación



In [5]:
total1 = tienda["Precio"].sum()
total2 = tienda2["Precio"].sum()
total3 = tienda3["Precio"].sum()
total4 = tienda4["Precio"].sum()

In [6]:
print(f"Facturación de la tienda 1: S/{total1:,.2f}")
print(f"Facturación de la tienda 2: S/{total2:,.2f}")
print(f"Facturación de la tienda 3: S/{total3:,.2f}")
print(f"Facturación de la tienda 4: S/{total4:,.2f}")

Facturación de la tienda 1: S/1,150,880,400.00
Facturación de la tienda 2: S/1,116,343,500.00
Facturación de la tienda 3: S/1,098,019,600.00
Facturación de la tienda 4: S/1,038,375,700.00


# 2. Ventas por categoría

In [ ]:
def dame_ventas_por_categoria(df, nombre_tienda):
    ventas=df.groupby("Categoría del Producto")["Precio"].sum()
    ventas = ventas.sort_index()

    print(f"Ventas por categoría en {nombre_tienda}: ")
    for categoria, total in ventas.items():
        print(f"{categoria}: S/{total:,.2f}")
    print()

In [ ]:
dame_ventas_por_categoria(tienda, "Tienda 1")
dame_ventas_por_categoria(tienda2, "Tienda 2")
dame_ventas_por_categoria(tienda3, "Tienda 3")
dame_ventas_por_categoria(tienda4, "Tienda 4")

Ventas por categoría en Tienda 1: 
Artículos para el hogar: S/12,698,400.00
Deportes y diversión: S/39,290,000.00
Electrodomésticos: S/363,685,200.00
Electrónicos: S/429,493,500.00
Instrumentos musicales: S/91,299,000.00
Juguetes: S/17,995,700.00
Libros: S/8,784,900.00
Muebles: S/187,633,700.00

Ventas por categoría en Tienda 2: 
Artículos para el hogar: S/14,746,900.00
Deportes y diversión: S/34,744,500.00
Electrodomésticos: S/348,567,800.00
Electrónicos: S/410,831,100.00
Instrumentos musicales: S/104,990,300.00
Juguetes: S/15,945,400.00
Libros: S/10,091,200.00
Muebles: S/176,426,300.00

Ventas por categoría en Tienda 3: 
Artículos para el hogar: S/15,060,000.00
Deportes y diversión: S/35,593,100.00
Electrodomésticos: S/329,237,900.00
Electrónicos: S/410,775,800.00
Instrumentos musicales: S/77,380,900.00
Juguetes: S/19,401,100.00
Libros: S/9,498,700.00
Muebles: S/201,072,100.00

Ventas por categoría en Tienda 4: 
Artículos para el hogar: S/15,074,500.00
Deportes y diversión: S/33,350,

# 3. Calificación promedio de la tienda


In [ ]:
calificacion_promedio1 = tienda["Calificación"].mean()
calificacion_promedio2 = tienda2["Calificación"].mean()
calificacion_promedio3 = tienda3["Calificación"].mean()
calificacion_promedio4 = tienda4["Calificación"].mean()

In [ ]:
print(f"Calificación promedio de la tienda 1: {calificacion_promedio1:,.2f}")
print(f"Calificación promedio de la tienda 2: {calificacion_promedio2:,.2f}")
print(f"Calificación promedio de la tienda 3: {calificacion_promedio3:,.2f}")
print(f"Calificación promedio de la tienda 4: {calificacion_promedio4:,.2f}")

Calificación promedio de la tienda 1: 3.98
Calificación promedio de la tienda 2: 4.04
Calificación promedio de la tienda 3: 4.05
Calificación promedio de la tienda 4: 4.00


# 4. Productos más y menos vendidos

In [ ]:
def top_productos(df, nombre_tienda):
    contador = df["Producto"].value_counts()

    top_mas = contador.head(3)
    top_menos = contador.tail(3)

    print(f"Ranking de {nombre_tienda}:")

    print("--Top 3 productos más vendidos--")
    for producto, cantidad in top_mas.items():
        print(f"{producto} : {cantidad} unidades")

    print("--Top 3 productos menos vendidos--")
    for producto, cantidad in top_menos.items():
        print(f"{producto} : {cantidad} unidades")

    print()

In [ ]:
top_productos(tienda, "Tienda 1")
top_productos(tienda2, "Tienda 2")
top_productos(tienda3, "Tienda 3")
top_productos(tienda4, "Tienda 4")

Ranking de Tienda 1:
--Top 3 productos más vendidos--
Microondas : 60 unidades
TV LED UHD 4K : 60 unidades
Armario : 60 unidades
--Top 3 productos menos vendidos--
Olla de presión : 35 unidades
Auriculares con micrófono : 33 unidades
Celular ABXY : 33 unidades

Ranking de Tienda 2:
--Top 3 productos más vendidos--
Iniciando en programación : 65 unidades
Microondas : 62 unidades
Batería : 61 unidades
--Top 3 productos menos vendidos--
Mesa de comedor : 34 unidades
Impresora : 34 unidades
Juego de mesa : 32 unidades

Ranking de Tienda 3:
--Top 3 productos más vendidos--
Kit de bancas : 57 unidades
Mesa de comedor : 56 unidades
Cama king : 56 unidades
--Top 3 productos menos vendidos--
Mochila : 36 unidades
Microondas : 36 unidades
Bloques de construcción : 35 unidades

Ranking de Tienda 4:
--Top 3 productos más vendidos--
Cama box : 62 unidades
Cubertería : 59 unidades
Dashboards con Power BI : 56 unidades
--Top 3 productos menos vendidos--
Guitarra acústica : 37 unidades
Armario : 34 un

# 5. Envío promedio por tienda

In [9]:
costoEnvio_promedio1 = tienda["Costo de envío"].mean()
costoEnvio_promedio2 = tienda2["Costo de envío"].mean()
costoEnvio_promedio3 = tienda3["Costo de envío"].mean()
costoEnvio_promedio4 = tienda4["Costo de envío"].mean()

In [10]:
print(f"Costo de envío promedio de la tienda 1: S/{costoEnvio_promedio1:,.2f}")
print(f"Costo de envío promedio de la tienda 2: S/{costoEnvio_promedio2:,.2f}")
print(f"Costo de envío promedio de la tienda 3: S/{costoEnvio_promedio3:,.2f}")
print(f"Costo de envío promedio de la tienda 4: S/{costoEnvio_promedio4:,.2f}")

Costo de envío promedio de la tienda 1: S/26,018.61
Costo de envío promedio de la tienda 2: S/25,216.24
Costo de envío promedio de la tienda 3: S/24,805.68
Costo de envío promedio de la tienda 4: S/23,459.46


# 6. Facturación - costo de envío por tienda

In [11]:
totalreal1 = tienda["Precio"].sum() - tienda["Costo de envío"].sum()
totalreal2 = tienda2["Precio"].sum() - tienda2["Costo de envío"].sum()
totalreal3 = tienda3["Precio"].sum() - tienda3["Costo de envío"].sum()
totalreal4 = tienda4["Precio"].sum() - tienda4["Costo de envío"].sum()

In [12]:
print(f"Facturación real de la tienda 1: S/{totalreal1:,.2f}")
print(f"Facturación real de la tienda 2: S/{totalreal2:,.2f}")
print(f"Facturación real de la tienda 3: S/{totalreal3:,.2f}")
print(f"Facturación real de la tienda 4: S/{totalreal4:,.2f}")

Facturación real de la tienda 1: S/1,089,502,500.00
Facturación real de la tienda 2: S/1,056,858,400.00
Facturación real de la tienda 3: S/1,039,503,000.00
Facturación real de la tienda 4: S/983,058,300.00


In [7]:
import plotly.express as px

In [13]:
labels = ["Tienda 1","Tienda 2", "Tienda 3", "Tienda 4"]
values = [totalreal1,totalreal2,totalreal3, totalreal4 ]

graficoPie = px.pie(names=labels, values=values, title="Participación de cada tienda")
graficoPie.show()

# 7. Ticket promedio por tienda

In [14]:
promedioreal1 = totalreal1 / len(tienda)
promedioreal2 = totalreal2 / len(tienda2)
promedioreal3 = totalreal3 / len(tienda3)
promedioreal4 = totalreal4 / len(tienda4)


In [15]:
print(f"Ticket promedio de la tienda 1: S/{promedioreal1:,.2f}")
print(f"Ticket promedio de la tienda 2: S/{promedioreal2:,.2f}")
print(f"Ticket promedio de la tienda 3: S/{promedioreal3:,.2f}")
print(f"Ticket promedio de la tienda 4: S/{promedioreal4:,.2f}")

Ticket promedio de la tienda 1: S/461,849.30
Ticket promedio de la tienda 2: S/448,011.19
Ticket promedio de la tienda 3: S/440,654.09
Ticket promedio de la tienda 4: S/416,903.44


In [16]:
import plotly.express as px

In [17]:
labels = ["Tienda 1", "Tienda 2", "Tienda 3", "Tienda 4"]
values = [promedioreal1, promedioreal2, promedioreal3, promedioreal4]

graficoBarras = px.bar(
    x=labels,
    y=values,
    text=values,
    title="Promedio de ticket por tienda",
    labels={"x": "Tienda", "y": "Promedio de ticket"},
    color=labels
)

graficoBarras.update_traces(texttemplate='$%{text:.2f}', textposition='outside')
graficoBarras.update_layout(
    yaxis_tickprefix="$",
    xaxis_title="Tiendas",
    yaxis_title="USD",
    yaxis_range=[350000,max(values)*1.1])

graficoBarras.show()

# 8. Ventas vs Cantidad de pedidos por tienda

In [18]:
ventas = [totalreal1, totalreal2, totalreal3, totalreal4]
ventas_escaladas = [v/100000 for v in ventas]
pedidos = [len(tienda), len(tienda2), len(tienda3), len(tienda4)]
tiendas = ["Tienda 1", "Tienda 2", "Tienda 3", "Tienda 4"]

In [19]:
import plotly.graph_objects as go

fig = go.Figure()

fig.add_trace(go.Bar(
    x=tiendas,
    y=ventas_escaladas,
    name="Monto de ventas",
    marker_color="green",
    text=[f"${v:,.2f}" for v in ventas_escaladas],
    textposition='outside'
))

fig.add_trace(go.Bar(
    x=tiendas,
    y=pedidos,
    name="Cantidad de pedidos",
    marker_color="orange",
    text=pedidos,
    textposition='outside'
))

fig.update_layout(
    barmode='group',
    title="Monto de ventas vs Cantidad de pedidos por tienda",
    xaxis_title="Tiendas",
    yaxis_title="Valores",
    yaxis_tickprefix="",
    uniformtext_minsize=8,
    uniformtext_mode='hide'
)

fig.show()

# 9. Análisis del desempeño geográfico

In [20]:
tienda["Tienda"] = "Tienda 1"
tienda2["Tienda"] = "Tienda 2"
tienda3["Tienda"] = "Tienda 3"
tienda4["Tienda"] = "Tienda 4"


ventas_geo = pd.concat([tienda, tienda2, tienda3, tienda4], ignore_index=True)

#ventas_geo.rename(columns={"lat": "Latitud", "lon": "Longitud"}, inplace=True)

fig = px.scatter_map(
    ventas_geo,
    lat="lat",
    lon="lon",
    color="Tienda",
    size="Precio",  # opcional: para reflejar monto
    hover_data=["Producto", "Precio","Tienda"],
    zoom=3,
    height=600,
    title="Ventas geolocalizadas por tienda"
)

# Usar mapa base de Mapbox (requiere token o usar estilo open-source)
fig.update_layout(mapbox_style="open-street-map")
fig.update_layout(margin={"r":0,"t":40,"l":0,"b":0})

fig.show()

# ANÁLISIS FINAL

La Tienda 1 es la que genera mayor utilidad seguida de cerca por la tienda 2 y 3.
Aunque la tienda 4 es la que genera menor utilidad, el porcentaje de utilidades ronda el 24%, siendo casi la cuarta parte de todos sus ingresos.
La tienda 1 es también la que peor promedio de calificación tiene por parte de los usuarios pero casi es una diferencia nula con respecto al resto de tiendas.
El ticket promedio también es más alto en la tienda 1, y más bajo en la tienda 4.

# RECOMENDACIÓN FINAL

Prescindir de la tienda 4, es dejar de percibir casi el 25% de sus ingresos. Prescindir de cualquiera de las tiendas, disminuriría en aproximadamente 25% sus ingresos totales.
Se recomienda no prescindir de ninguna tienda pues todas suman positivamente en sus ingresos, pero si es estrictamente necesario dejar de contar con una tienda, la elegida sería la tienda 4 con la pérdida de ingresos ya comentada.
Se recomienda aplicar medidas para mejorar la calificación de los clientes con respecto a cada tienda.
Se recomienda aplicar medidad para mejorar el proceso de envío y minimizar esos costos,